### PR TITLE
update package scripts for easier maintenance

### PR DESCRIPTION
**Pull request by delphiactual**
based off of prettier/prettier/package.json

### CODE REVIEW

File: .github/workflows/pr-build.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ jobs:
       - name: Lint check
         run: |
           find . -type d -name "node_modules" -prune -o -type f -iregex '.*.ts\|.*.js\|.*.tsx\|.*.jsx' -exec ./build/set-mtime-to-md5.sh {} \;
-          yarn run lint-check
+          yarn check:eslint
```
Comment (bhollis) [inline .github/workflows/pr-build.yml:69]: Don't we want to run all lints?

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ Here are some tips to make sure your pull request can be merged smoothly:
 1. Resist the temptation to change more than one thing in your PR. Keeping PRs focused on a single change makes them much easier to review and accept. If you want to change multiple things, or clean up/refactor the code, make a new branch and submit those changes as a separate PR.
 1. All of our code is written in [TypeScript](https://typescriptlang.org) and uses React to build UI components..
 1. Take advantage of the [native JavaScript Array methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) and the [Lodash](https://lodash.com/) library to write compact, easy-to-understand code.
-1. Be sure to run `yarn run lint` before submitting your PR - it'll catch most style problems and make things much easier to merge.
+1. Be sure to run `yarn lint fix` before submitting your PR - it'll catch most style problems and make things much easier to merge.
```
Comment (bhollis) [inline docs/CONTRIBUTING.md:9]: Wouldn't this be `yarn fix`?

File: package.json [hunk 3]
```diff
@@ -9,8 +9,16 @@
...
   "scripts": {
     "test": "cross-env NODE_ENV=test jest --verbose",
-    "lint": "jsonsort src/locale && eslint --fix src --ext .js,.ts,.tsx && stylelint --fix \"src/**/*.scss\"",
-    "lint-check": "eslint src --ext .js,.ts,.tsx --cache --cache-location .eslintcache",
+    "check:eslint": "eslint src --ext .js,.ts,.tsx --cache --cache-location .eslintcache",
+    "lint": "run-p lint:*",
+    "lint:eslint": "eslint src --ext .js,.ts,.tsx",
+    "lint:prettier": "prettier src/**/*.{js,ts,tsx,scss,html} --check",
+    "lint:stylelint": "stylelint \"src/**/*.scss\"",
+    "fix": "run-s fix:eslint fix:json fix:stylelint fix:prettier",
```
Comment (bhollis) [inline package.json:17]: what's the difference between run-s and run-p? Can this be `run-s fix:*`?

**Review by bhollis — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (delphiactual) [reply package.json:17]: run-p is run in parallel, run-s is sequential

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (delphiactual) [reply docs/CONTRIBUTING.md:9]: yep, good catch

File: .github/workflows/pr-build.yml (hunk 1 above)
Comment (delphiactual) [reply .github/workflows/pr-build.yml:69]: we never did before this was the lint with cache

**Review by nev-r — APPROVED** [review]

**Pull request merged**